### DISCUSSION TITLE
Rails Data Export

### DISCUSSION THREAD

**Discussion by leoluchini** [open-telemetry/opentelemetry-ruby-contrib#830] [Q&A]
Hi everyone,

I'm currently working on implementing `opentelemetry-instrumentation-rails`, and while it is functional, I would like to modify the way the data is exported.

Currently, when I check the dashboard with the data, it is grouped by controller name and action, as shown in the attached screenshot:

![Screenshot 2024-01-23 at 08 51 26](https://github.com/open-telemetry/opentelemetry-ruby-contrib/assets/5969628/177b22cc-9299-496a-9f1d-69544a9fff59)

![Screenshot 2024-01-23 at 08 54 01](https://github.com/open-telemetry/opentelemetry-ruby-contrib/assets/5969628/bacc2c46-e35a-41e6-8810-2b263bb2794e)

Instead of displaying the controller name and action, I would prefer to have the HTTP action and the path, such as `GET /ping`. Is this customization possible? Any assistance would be greatly appreciated.

This is my rails configuration

```
OpenTelemetry::SDK.configure do |c|
  c.service_name = ENV.fetch('OTEL_SERVICE_NAME')
  c.use_all('OpenTelemetry::Instrumentation::Rack' => { url_quantization: lambda { |path, _env| path.to_s } })
end
```

Feel free to let me know if you have any specific preferences or if there are additional details you'd like to include in the message.

**Comment by arielvalentin** [comment]
Hello @leoluchini 

This is something that we'd also like to support. 

Would you be interested in helping us implement this?

**Comment by leoluchini** [comment]
Hi @arielvalentin

Sure! Do you have a requirement or ticket to guide me?

**Reply by arielvalentin** [reply]
Not yet, but I can write one up!

We have the community meeting Tuesday's from 11am-12pm Central Time US. You are also welcome to join to chat about it in detail.

**Reply by leoluchini** [reply]
Ok, let me know when you have it and I'll be happy to contribute.

Regarding the community meeting send me the invite and I will join it.

Thanks

**Reply by arielvalentin** [reply]
The Google calendar and meetings are open to the public. For more information, take a look at the following:

https://opentelemetry.io/community/

https://github.com/open-telemetry/community

If you haven't already, please consider joining the CNCF slack for more near-realtime communication: https://communityinviter.com/apps/cloud-native/cncf
